File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/command/xml/insert/Delete.java
package com.kite9.k9server.command.xml.insert;

import org.w3c.dom.Element;

import com.kite9.k9server.adl.holder.pipeline.ADLDom;
import com.kite9.k9server.command.CommandException;

/**
 * Delete is basically the opposite of insert; you have to say what you're deleting so it can be
 * checked/undone.
 * 
 * @author robmoffat
 *
 */
public class Delete extends AbstractInsertCommand {

	public String base64Element;
	
	@Override
	public Mismatch applyCommand(ADLDom in) throws CommandException {
		checkProperties();
		return doDelete(in);
	}

	@Override
	public Mismatch undoCommand(ADLDom in) throws CommandException {
		checkProperties();
		return doInsert(in);
	}

	@Override
	protected Element getContents(ADLDom in) {
		return copyWithoutContainedIds(decodeElement(base64Element, in));
	}

	@Override
	protected void checkProperties() {
		super.checkProperties();
		ensureNotNull("base64Element", fragmentId);
	}

}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/pipeline/command/xml/insert/Delete.kt
package com.kite9.pipeline.command.xml.insert

import com.kite9.pipeline.adl.holder.pipeline.ADLDom
import com.kite9.pipeline.command.Command.Mismatch
import com.kite9.pipeline.command.CommandContext
import org.w3c.dom.Element

/**
 * Delete is basically the opposite of insert; you have to say what you're deleting so it can be
 * checked/undone.
 *
 * @author robmoffat
 */
class Delete : AbstractInsertCommand() {

	var base64Element: String = ""

    override fun applyCommand(d: ADLDom, ctx: CommandContext): Mismatch? {
        checkProperties()
        return doDelete(d, ctx)
    }

    override fun undoCommand(d: ADLDom, ctx: CommandContext): Mismatch? {
        checkProperties()
        return doInsert(d, ctx)
    }

    override fun getContents(d: ADLDom, ctx: CommandContext): Element? {
        return copyWithoutContainedIds(ctx.decodeElement(base64Element, d))
    }

    override fun checkProperties() {
        super.checkProperties()
        ensureNotNull("base64Element", fragmentId)
    }
}